### PR TITLE
CICD: Macos11 support

### PR DESCRIPTION
**Pull request by algojack**
* added MacOS 11 to nightly jobs (Darwin ARM_64)
* renewed cache version due to it's not backwards compatible with macos 11

Tested by running in circleci: https://app.circleci.com/pipelines/github/algorand/go-algorand/8694/workflows/2e801391-7eaa-4f5f-a522-9ad05b49326c

### CODE REVIEW

**Review by algolucky — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -673,7 +673,7 @@ jobs:
       - run:
           no_output_timeout: 45m
           command: |
-            #export PATH=$(echo "$PATH" | sed -e 's|:/home/circleci/\.go_workspace/bin||g' | sed -e 's|:/usr/local/go/bin||g')
+            # export PATH=$(echo "$PATH" | sed -e 's|:/home/circleci/\.go_workspace/bin||g' | sed -e 's|:/usr/local/go/bin||g')
```
Comment (egieseke) [inline .circleci/config.yml:676]: why is this here?

**Review by egieseke — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (algojack) [reply .circleci/config.yml:676]: because ide added a space

Comment (algojack) [reply .circleci/config.yml:676]: since we don't currently run that workflow (windows build) I can't say for sure, and don't want to remove it until we test if we need it or not.

**Review by egieseke — APPROVED** [review]

**Review by algolucky — APPROVED** [review]
